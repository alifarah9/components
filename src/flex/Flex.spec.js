import React from 'react';
import serializer from 'jest-emotion';
/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';

import Flex from './';

const Direction = { xs: 'row', sm: 'row-reverse', md: 'column', lg: 'column-reverse', xl: 'row' };

const customMediaQueries = [0, 10, 20, 30, 40].map(bp => `@media (min-width: ${bp}px)`);

expect.addSnapshotSerializer(serializer);

const props = {
  direction: Direction,
  marginX: 0,
  marginY: 1,
  paddingX: 2,
  paddingY: 3,
};

describe('Box', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Flex {...props}>
        <div>This is a children.</div>
      </Flex>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('renders correctly with custom media queries', () => {
    const component = renderer.create(
      <Flex {...props} customMediaQueries={customMediaQueries}>
        <div>This is a children.</div>
      </Flex>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
