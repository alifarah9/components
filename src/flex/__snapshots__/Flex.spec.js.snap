// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
@media (min-width:0px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:576px) {
  .emotion-0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:768px) {
  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:992px) {
  .emotion-0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  className="d-flex emotion-0"
>
  <div
    marginX={0}
    marginY={1}
    paddingX={2}
    paddingY={3}
  >
    This is a children.
  </div>
</div>
`;

exports[`Box renders correctly with custom media queries 1`] = `
@media (min-width:0px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:10px) {
  .emotion-0 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }
}

@media (min-width:20px) {
  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:30px) {
  .emotion-0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media (min-width:40px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  className="d-flex emotion-0"
>
  <div
    marginX={0}
    marginY={1}
    paddingX={2}
    paddingY={3}
  >
    This is a children.
  </div>
</div>
`;
